src/Features/VisualBasic/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.vb

```diff
@@ -1,250 +1,24 @@
         End Function
 
-        Private Function GetInvertedIfNode(
-            ifNode As SingleLineIfStatementSyntax,
-            document As Document,
-            generator As SyntaxGenerator,
-            syntaxFacts As ISyntaxFactsService,
-            semanticModel As SemanticModel,
-            cancellationToken As CancellationToken) As SingleLineIfStatementSyntax
-
-            Dim elseClause = ifNode.ElseClause
-
-            ' If we're moving a single line if from the else body to the if body,
-            ' and it is the last statement in the body, we have to introduce an extra
-            ' StatementTerminator Colon and Else token.
-            Dim newIfStatements = elseClause.Statements
-
-            If newIfStatements.Count > 0 Then
-                newIfStatements = newIfStatements.Replace(
-                    newIfStatements.Last,
-                    newIfStatements.Last.WithTrailingTrivia(elseClause.ElseKeyword.GetPreviousToken().TrailingTrivia))
-            End If
-
-            If elseClause.Statements.Count > 0 AndAlso
-               elseClause.Statements.Last().Kind = SyntaxKind.SingleLineIfStatement Then
-
-                Dim singleLineIf = DirectCast(elseClause.Statements.Last, SingleLineIfStatementSyntax)
-
-                ' Create an Extra 'Else'
-                If singleLineIf.ElseClause Is Nothing Then
-
-                    ' Replace the last EOL of the IfPart with a :
-                    Dim trailing = singleLineIf.GetTrailingTrivia()
-                    If trailing.Any(SyntaxKind.EndOfLineTrivia) Then
-                        Dim eol = trailing.Last(Function(t) t.Kind = SyntaxKind.EndOfLineTrivia)
-                        trailing = trailing.Select(Function(t) If(t = eol, SyntaxFactory.ColonTrivia(syntaxFacts.GetText(SyntaxKind.ColonTrivia)), t)).ToSyntaxTriviaList()
-                    End If
-
-                    Dim withElsePart = singleLineIf.WithTrailingTrivia(trailing).WithElseClause(
-                        SyntaxFactory.SingleLineElseClause(SyntaxFactory.List(Of StatementSyntax)()))
-
-                    ' Put the if statement with the else into the statement list
-                    newIfStatements = elseClause.Statements.Replace(elseClause.Statements.Last, withElsePart)
-                End If
-            End If
-
-            Return ifNode.WithCondition(DirectCast(Negate(ifNode.Condition, generator, syntaxFacts, semanticModel, cancellationToken), ExpressionSyntax)) _
-                         .WithStatements(newIfStatements) _
-                         .WithElseClause(elseClause.WithStatements(ifNode.Statements).WithTrailingTrivia(elseClause.GetTrailingTrivia()))
+        Protected Overrides Function GetAnalyzer(ifStatement As SyntaxNode) As IAnalyzer
+            Return If(TypeOf ifStatement Is SingleLineIfStatementSyntax,
+                      DirectCast(New SingleLineIfStatementAnalyzer, IAnalyzer),
```

Comment:
oh, that's basically what you do :)   I guess my only feedback would be have two different VisualBasicInvertIfCodeRefactoringProvider providers.  One for each type.  (Just a thought)

Reply:
(resolved)